runtime/src/main/java/io/novafoundation/nova/runtime/network/updaters/BlockNumberUpdater.kt

```diff
@@ -1,24 +1,49 @@
 package io.novafoundation.nova.runtime.network.updaters
 
+import android.util.Log
 import io.novafoundation.nova.common.utils.Modules
 import io.novafoundation.nova.common.utils.system
 import io.novafoundation.nova.core.storage.StorageCache
-import io.novafoundation.nova.core.updater.GlobalScope
+import io.novafoundation.nova.core.updater.EmptyScope
+import io.novafoundation.nova.core.updater.SharedRequestsBuilder
+import io.novafoundation.nova.core.updater.UpdateScope
+import io.novafoundation.nova.core.updater.Updater
 import io.novafoundation.nova.runtime.multiNetwork.ChainRegistry
-import io.novafoundation.nova.runtime.state.SelectedAssetOptionSharedState
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import io.novafoundation.nova.runtime.multiNetwork.getRuntime
 import io.novasama.substrate_sdk_android.runtime.RuntimeSnapshot
 import io.novasama.substrate_sdk_android.runtime.metadata.storage
 import io.novasama.substrate_sdk_android.runtime.metadata.storageKey
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.emptyFlow
+import kotlinx.coroutines.flow.onEach
 
 class BlockNumberUpdater(
-    chainRegistry: ChainRegistry,
-    crowdloanSharedState: SelectedAssetOptionSharedState<*>,
-    storageCache: StorageCache
-) : SingleStorageKeyUpdater<Unit>(GlobalScope, crowdloanSharedState, chainRegistry, storageCache) {
+    private val chainRegistry: ChainRegistry,
+    private val storageCache: StorageCache
+) : Updater<Chain> {
```

Comment:
Am I right that you removed SingleStorageKeyUpdater inheritance because it requires you to pass a chain id?
Thats actually deprecated since chain is now always available in `listenForUpdates`

Ideally we should remove it from SingleStorageKeyUpdater  but given that we have many updaters that inherts from it, there will be a lot of extra work to refact it now
So my suggestion is to make `chainIdHolder` filed optional in SingleStorageKeyUpdater   constructor and mark it as deprecated. This way, BlockNumberUpdater could just ignore it

Reply:
Not really right. I just renamed old BlockNumberUpdater to SharedAssetBlockNumberUpdater and created BlockNumberUpdater that does not depend on the external state.

I think that this is not a very correct diff)

Reply:
Okay I see

My suggestion seems to still be relevant though - instead of dublicating logic of  SingleStorageKeyUpdater  we could deprecate its "state" since chainIdHolder is no longer actually needed and use SingleStorageKeyUpdater   in a stateless way

Reply:
I think we can create a task and refactor it later. I think it's a minor issue